blockmanager.go

```diff
@@ -402,15 +402,21 @@ func (b *blockManager) cfHandler() {
 		b.wg.Done()
 	}()
 
-	log.Infof("Waiting for block headers to sync, then will start " +
-		"cfheaders sync...")
+waitForHeaders:
+	// We'll wait until the main header sync is either finished or the
+	// filter headers are lagging at least a checkpoint interval behind the
+	// block headers, before we actually start to sync the set of
+	// cfheaders. We do this to speed up the sync, as the check pointed
+	// sync is faster, than fetching each header from each peer during the
+	// normal "at tip" syncing.
+	//
+	// NOTE: We can grab the filterHeaderTip here without a lock, as this
+	// is the only goroutine that can modify this value.
+	log.Infof("Waiting for block headers to sync, then will start "+
+		"cfheaders sync from height %v...", b.filterHeaderTip)
 
-	// We'll wait until the main header sync is mostly finished before we
-	// actually start to sync the set of cfheaders. We do this to speed up
-	// the sync, as the check pointed sync is faster, then fetching each
-	// header from each peer during the normal "at tip" syncing.
 	b.newHeadersSignal.L.Lock()
-	for !b.IsCurrent() {
+	for !(b.IsCurrent() || b.filterHeaderTip+wire.CFCheckptInterval <= b.headerTip) {
```

Comment:
already discussed irl, but it might be worth reversing the order (again) to avoid db hits amap

Reply:
reversed.